backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/scheduling/utils/VdsCpuUnitPinningHelper.java

```diff
@@ -101,14 +104,17 @@ private void previewPinOfPendingExclusiveCpus(List<VdsCpuUnit> cpuTopology, Map<
      * @param host VDS object.
      * @return List<{@link VdsCpuUnit}>. The list of VdsCpuUnit we are going to use. If not possible, return an empty List.
      */
-    public List<VdsCpuUnit> allocateDedicatedCpus(VM vm, Map<Guid, List<VdsCpuUnit>> vmToPendingPinnings, VDS host) {
+    public List<VdsCpuUnit> allocateCpus(VM vm, Map<Guid, List<VdsCpuUnit>> vmToPendingPinnings, VDS host) {
         if (vm.getCpuPinningPolicy() == CpuPinningPolicy.NONE) {
             return new ArrayList<>();
         }
         List<VdsCpuUnit> cpuTopology = resourceManager.getVdsManager(host.getId()).getCpuTopology();
         if (cpuTopology.isEmpty()) {
             return new ArrayList<>();
         }
+        if (vm.getCpuPinningPolicy() == CpuPinningPolicy.RESIZE_AND_PIN_NUMA) {
+            vmHandler.updateCpuAndNumaPinning(vm, host.getId());
+        }
```

Comment:
this is not something I would expect this method to do because it's beyond allocating the CPUs
how about extracting the code out of this method to SchedulingManager and rename this method to something like updatePhysicalCpuAllocations?

and how do we prevent the topology from changing when scheduling a vm as part of migrate-vm?

Reply:
> this is not something I would expect this method to do because it's beyond allocating the CPUs how about extracting the code out of this method to SchedulingManager and rename this method to something like updatePhysicalCpuAllocations?
> 
Why renaming? it sets the VM CPU topology and NUMA.

> and how do we prevent the topology from changing when scheduling a vm as part of migrate-vm?
Good point. Need to check that (maybe check the last VM status).

Reply:
> > this is not something I would expect this method to do because it's beyond allocating the CPUs how about extracting the code out of this method to SchedulingManager and rename this method to something like updatePhysicalCpuAllocations?
> 
> Why renaming? it sets the VM CPU topology and NUMA.

I meant the method that was named `allocateDedicatedCpus` and now changes to `allocateCpus` - as it doesn't allocate CPUs in case of the MANUAL policy - the CPUs were allocated by the user and we just update the physical topology with these allocations (IIRC at the beginning this method only handled the dedicated policy so the name made sense)